doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/event-listeners-timing-con.adoc

```diff
@@ -0,0 +1,21 @@
+[id='event-listeners-timing-con_{context}']
+= Timing of calls to event listeners
+
+A number of event listener calls are `before` and `after` events, for example, `beforeNodeLeft()` and `afterNodeLeft()`, `beforeTaskActivatedEvent()` and `afterTaskActivatedEvent()`.
+
+The `before` and `after` event calls typically act like a stack. If event A directly causes event B, the following sequence of calls happens:
+
+* Before A
+* Before B
+* After B
+* After A
+
+For example, if leaving node X triggers node Y, all event calls related to triggering node Y occur between the `beforeNodeLeft` and `afterNodeLeft` calls for node X.
```

Comment:
Just wondering why you switched to X and Y. Wouldn't it make more sense to stay with A and B? For example, if leaving node A....

Reply:
A and B are events. X and Y are nodes.

Reply:
Right! Thanks.